sdk/go/pulumi/run.go

```diff
@@ -81,7 +84,12 @@ func RunErr(body RunFunc, opts ...RunOption) error {
 	}
 	defer contract.IgnoreClose(ctx)
 
-	return RunWithContext(ctx, body)
+	err = RunWithContext(ctx, body)
```

Comment:
It's a bit concerning that this change is editing RunErr and makes it (1) log the error to ctx.Log.Error and (2) also panic in contract.AssertNoError. Why is this needed? Do we understand how RunErr is used in the wild? It does not seem to be internal-only?

A more minimal change would only edit `Run`, that one feels safer to edit , that is at this point if we are doing this:

```
			fmt.Fprintf(os.Stderr, "error: program failed: %v\n", err)
			os.Exit(1)
```

Then this function is unlikely to be used anywhere except for toplevel program entry point (it does os.Exit), so we may edit this here to send a message to ctx.Log instead of stdout, and then os.Exit(1) as before.

Reply:
I'll move my changes from `RunErr` to `Run` and then remove the `AssertNoError` check for `ctx.Log.Error`